src/test/test-notify-recv.c

```diff
@@ -40,10 +48,13 @@ static int on_recv(sd_event_source *s, int fd, uint32_t revents, void *userdata)
                 ASSERT_STREQ(l[0], "FIRST_MESSAGE=1");
                 ASSERT_NULL(l[1]);
                 ASSERT_EQ(c->data, 0u);
+                ASSERT_NULL(fds);
         } else if (strv_contains(l, "SECOND_MESSAGE=1")) {
                 ASSERT_STREQ(l[0], "SECOND_MESSAGE=1");
                 ASSERT_STREQ(l[1], "ADDITIONAL_DATA=hoge");
                 ASSERT_EQ(c->data, 1u);
+                ASSERT_NOT_NULL(fds);
+                ASSERT_EQ(fdset_size(fds), 2u);
```

Comment:
Nice.

Could you check that the passed fds are actually sent by the parent? Something like the following:
```c
int i;
FDSET_FOREACH(i, fds) {
        _cleanup_free_ char *path = NULL;
        ASSERT_OK(fd_get_path(i, &path));
        ASSERT_TRUE(STR_IN_SET(path, "/tmp", "/dev/null"));
}
```